web/frontend/src/components/do-command/index.svelte

```diff
@@ -0,0 +1,113 @@
+<script lang="ts">
+
+import { type Client, commonApi } from '@viamrobotics/sdk';
+import { toast } from '@/lib/toast';
+import { resourceNameToString } from '@/lib/resource';
+import { doCommand } from '@/api/do-command';
+import Collapse from '@/components/collapse.svelte';
+
+export let resources: commonApi.ResourceName.AsObject[];
+export let client: Client;
+
+let selectedComponent = '';
+let input = '';
+let output = '';
+let executing = false;
+
+const handleDoCommand = async (name: string, command: string) => {
+  if (!name || !command) {
+    return;
+  }
+
+  executing = true;
+
+  try {
+    const outputObject = await doCommand(client, name, command);
+
+    if (outputObject) {
+      output = JSON.stringify(outputObject, null, '\t');
+    } else {
+      toast.error(`Invalid response when executing command on ${name}`);
+    }
+  } catch (error) {
+    toast.error(`Error executing command on ${name}: ${error}`);
+  }
+
+  executing = false;
+};
+
+const handleSelectComponent = (event: CustomEvent) => {
+  selectedComponent = event.detail.value;
+};
+
+const handleEditorInput = (event: CustomEvent) => {
+  input = event.detail.value;
+};
+
+const namesToPrettySelect = (resourcesToPretty: commonApi.ResourceName.AsObject[]): string => {
+  const simple = new Map<string, number>();
+
+  for (const resource of resourcesToPretty) {
+    if (!simple.has(resource.name)) {
+      simple.set(resource.name, 0);
+    }
+    simple.set(resource.name, simple.get(resource.name)! + 1);
+  }
+
+  return resourcesToPretty.map((res) => {
+    if (simple.get(res.name) === 1) {
+      return res.name;
+    }
+    return resourceNameToString(res);
+  }).join(',');
+};
+
+</script>
+
+<Collapse title="DoCommand()">
+  <div class="h-full w-full border border-t-0 border-medium p-4">
+    <v-select
+      label="Selected Component"
+      placeholder="Select a component"
+      options={namesToPrettySelect(resources)}
+      value={selectedComponent}
+      disabled={executing ? 'true' : 'false'}
```

Comment:
Is there a reason these are strings and not boolean ?

Reply:
because they're html attributes :|

Reply:
ah i was thinking we could get around that with Svelte; guess not